### PR TITLE
Audit fixes and meta transaction support for RootChainManage, root tokens

### PR DESCRIPTION
**Pull request by sanchaymittal**
- Addresses exhibit 15, 16 from Audit report
  - [x] Exhibit 15 - Using same chainId for child token contracts is exploitable. Changed NetworkAgnostic feature to native meta transaction. `chainid` opcode is used so it is always the native chain id. This chain id is used as salt in EIP712 domain separator so that metamask allows signing tx with different chain id than currently selected network.
  - [x] Exhibit 16 - `executeMetaTransaction` function does not check the address so there is possibility to replay attack for different contract. Added a test to verify this. `verifyingContract` address is part of the signature so replay attack is not successful.

- Also adds meta transaction support to `RootChainManager`, and all root tokens. Since `RootChainManager` follows the upgradeable pattern, EIP712Base is modified to use initialize function instead of doing it in constructor.
- Updates evmVersion to istanbul to support `chainId` opcode. Matic test evm is also updated to support this.
- Adds multiple other tests for meta transactions.
- Adds check for signer address zero in meta transaction.
- Rearranges steps in initializer modifier so both derived and base contract can use initializer.

### CODE REVIEW

File: flat/ChildChainManager.sol [hunk 1]
```diff
@@ -108,8 +108,8 @@ contract Initializable {
 
     modifier initializer() {
         require(!inited, "already inited");
-        inited = true;
         _;
+        inited = true;
```
Comment (jdkanani) [inline flat/ChildChainManager.sol:112]: Why changed the order? We should set first just to avoid reentrancy.

File: flat/ChildERC1155.sol [hunk 2]
```diff
@@ -1510,12 +1510,26 @@ interface IChildToken {
...
+
+pragma solidity 0.6.6;
+
+contract Initializable {
+    bool inited = false;
+
+    modifier initializer() {
+        require(!inited, "already inited");
+        _;
+        inited = true;
```
Comment (jdkanani) [inline flat/ChildERC1155.sol:1523]:
```suggestion
        inited = true;
        _;
```

File: flat/ChildERC20.sol [hunk 3]
```diff
@@ -1214,12 +1214,26 @@ interface IChildToken {
...
+
+pragma solidity 0.6.6;
+
+contract Initializable {
+    bool inited = false;
+
+    modifier initializer() {
+        require(!inited, "already inited");
+        _;
+        inited = true;
```
Comment (jdkanani) [inline flat/ChildERC20.sol:1227]:
```suggestion
        inited = true;
        _;
```

File: flat/ChildERC721.sol [hunk 4]
```diff
@@ -1876,12 +1876,26 @@ interface IChildToken {
...
+
+pragma solidity 0.6.6;
+
+contract Initializable {
+    bool inited = false;
+
+    modifier initializer() {
+        require(!inited, "already inited");
+        _;
+        inited = true;
```
Comment (jdkanani) [inline flat/ChildERC721.sol:1889]:
```suggestion
        inited = true;
        _;
```

File: flat/ChildMintableERC721.sol [hunk 5]
```diff
@@ -1876,12 +1876,26 @@ interface IChildToken {
...
+
+pragma solidity 0.6.6;
+
+contract Initializable {
+    bool inited = false;
+
+    modifier initializer() {
+        require(!inited, "already inited");
+        _;
+        inited = true;
```
Comment (jdkanani) [inline flat/ChildMintableERC721.sol:1889]:
```suggestion
        inited = true;
        _;
```

File: flat/DummyERC1155.sol [hunk 6]
```diff
@@ -1017,12 +1017,26 @@ contract ERC1155 is Context, ERC165, IERC1155, IERC1155MetadataURI {
...
+
+pragma solidity 0.6.6;
+
+contract Initializable {
+    bool inited = false;
+
+    modifier initializer() {
+        require(!inited, "already inited");
+        _;
+        inited = true;
```
Comment (jdkanani) [inline flat/DummyERC1155.sol:1030]:
```suggestion
        inited = true;
        _;
```

File: flat/ChildChainManager.sol (hunk 1 above)
Comment (arthcp) [reply flat/ChildChainManager.sol:112]: `RootChainManager` inherits `EIP712Base`, both are initializable.
RootChainManager initializer calls EIP712Base initializer.
Both these check the same `inited` variable. 
If `inited` is set before, then check fails in `EIP712Base`.

Comment (arthcp) [reply flat/ChildChainManager.sol:112]: These functions will be called only by us while deploying, can we assume it safe?

**Review by jdkanani — APPROVED** [review]

**Pull request merged**